In [1]:
import pandas as pd

In [2]:
import numpy as np

In [3]:
import matplotlib.pyplot as plt

In [4]:
import seaborn as sns

In [5]:
gdp_df = pd.read_csv(r'C:\\Users\\J2das\\Documents\\NSS\\PYTHON\\Projects\\gdp-and-internet-usage-j2das1\\data\\gdp_percapita.csv', nrows=7663)

In [6]:
gdp_df

Country or Area  Year        Value  Value Footnotes
0        Afghanistan  2021  1517.016266              NaN
1        Afghanistan  2020  1968.341002              NaN
2        Afghanistan  2019  2079.921861              NaN
3        Afghanistan  2018  2060.698973              NaN
4        Afghanistan  2017  2096.093111              NaN
...              ...   ...          ...              ...
7657        Zimbabwe  1994  2670.106615              NaN
7658        Zimbabwe  1993  2458.783255              NaN
7659        Zimbabwe  1992  2468.278257              NaN
7660        Zimbabwe  1991  2781.787843              NaN
7661        Zimbabwe  1990  2704.757299              NaN

[7662 rows x 4 columns]

In [7]:
gdp_df.shape

(7662, 4)

Question 5: How many rows and columns are in `gdp_df`? What are the data types of each column? 7662 rows and 4 colums


6. Drop the `Value Footnotes` column and rename the remaining three to 'Country', 'Year', and 'GDP_Per_Capita'.

In [10]:
gdp_df = gdp_df.drop(columns = ['Value Footnotes'])

In [11]:
print(gdp_df)

     Country or Area  Year        Value
0        Afghanistan  2021  1517.016266
1        Afghanistan  2020  1968.341002
2        Afghanistan  2019  2079.921861
3        Afghanistan  2018  2060.698973
4        Afghanistan  2017  2096.093111
...              ...   ...          ...
7657        Zimbabwe  1994  2670.106615
7658        Zimbabwe  1993  2458.783255
7659        Zimbabwe  1992  2468.278257
7660        Zimbabwe  1991  2781.787843
7661        Zimbabwe  1990  2704.757299

[7662 rows x 3 columns]


In [12]:
gdp_df = gdp_df.rename(columns = {'Country or Area':'Country','Value':'GDP_Per_Capita'})

In [13]:
print(gdp_df)

          Country  Year  GDP_Per_Capita
0     Afghanistan  2021     1517.016266
1     Afghanistan  2020     1968.341002
2     Afghanistan  2019     2079.921861
3     Afghanistan  2018     2060.698973
4     Afghanistan  2017     2096.093111
...           ...   ...             ...
7657     Zimbabwe  1994     2670.106615
7658     Zimbabwe  1993     2458.783255
7659     Zimbabwe  1992     2468.278257
7660     Zimbabwe  1991     2781.787843
7661     Zimbabwe  1990     2704.757299

[7662 rows x 3 columns]


7. How many countries have data for all years? Which countries are missing many years of data? Look at the number of observations per year. What do you notice?

7a. How many countries have data for all years?

In [16]:
gdp_country = gdp_df.groupby('Country')

In [17]:
gdp_country.count()

Year  GDP_Per_Capita
Country                                          
Afghanistan                    20              20
Africa Eastern and Southern    33              33
Africa Western and Central     33              33
Albania                        33              33
Algeria                        33              33
...                           ...             ...
Viet Nam                       33              33
West Bank and Gaza             29              29
World                          33              33
Zambia                         33              33
Zimbabwe                       33              33

[242 rows x 2 columns]

In [18]:
total_count = gdp_country.count().sort_values(by = 'Year')

In [19]:
print(total_count)

                           Year  GDP_Per_Capita
Country                                        
Djibouti                     10              10
Somalia                      10              10
Turks and Caicos Islands     12              12
Sint Maarten (Dutch part)    14              14
Kosovo                       15              15
...                         ...             ...
Jamaica                      33              33
Japan                        33              33
Jordan                       33              33
IDA total                    33              33
Zimbabwe                     33              33

[242 rows x 2 columns]


In [20]:
all_years_count = total_count.loc[total_count.Year == 33]

In [21]:
print(all_years_count)

                             Year  GDP_Per_Capita
Country                                          
Papua New Guinea               33              33
Pakistan                       33              33
Australia                      33              33
Pacific island small states    33              33
Austria                        33              33
...                           ...             ...
Jamaica                        33              33
Japan                          33              33
Jordan                         33              33
IDA total                      33              33
Zimbabwe                       33              33

[202 rows x 2 columns]


In [22]:
all_years_count.Year.value_counts()

Year
33    202
Name: count, dtype: int64

7b.  Which countries are missing many years of data?

In [24]:
total_count.head(10)

Year  GDP_Per_Capita
Country                                        
Djibouti                     10              10
Somalia                      10              10
Turks and Caicos Islands     12              12
Sint Maarten (Dutch part)    14              14
Kosovo                       15              15
Cayman Islands               16              16
Nauru                        19              19
Afghanistan                  20              20
Curaçao                     22              22
Palau                        22              22

7c. Look at the number of observations per year. What do you notice? when GDP_per_capita is low the country will be missing lots of years of data

In [26]:
gdp_df.groupby(['Country']).agg({'Year' : ['count','mean','max','min'],'GDP_Per_Capita' :['count','mean','max','min']})

Year                     GDP_Per_Capita  \
                            count    mean   max   min          count   
Country                                                                
Afghanistan                    20  2011.5  2021  2002             20   
Africa Eastern and Southern    33  2006.0  2022  1990             33   
Africa Western and Central     33  2006.0  2022  1990             33   
Albania                        33  2006.0  2022  1990             33   
Algeria                        33  2006.0  2022  1990             33   
...                           ...     ...   ...   ...            ...   
Viet Nam                       33  2006.0  2022  1990             33   
West Bank and Gaza             29  2008.0  2022  1994             29   
World                          33  2006.0  2022  1990             33   
Zambia                         33  2006.0  2022  1990             33   
Zimbabwe                       33  2006.0  2022  1990             33   

                                                                      
                                     mean           max          min  
Country                                                               
Afghanistan                   1790.659581   2165.340915  1260.060581  
Africa Eastern and Southern   3238.934911   3661.360566  2715.131116  
Africa Western and Central    3337.603210   4148.547272  2561.665446  
Albania                       8674.425710  15501.662931  3264.820757  
Algeria                      10058.147899  11888.322967  7869.270272  
...                                   ...           ...          ...  
Viet Nam                      5710.759399  11396.531469  2099.394649  
West Bank and Gaza            5110.371012   6438.933640  3395.855028  
World                        12962.677329  17485.934316  9665.890260  
Zambia                        2674.800172   3425.948936  1991.185925  
Zimbabwe                      2301.826155   2931.725144  1353.207369  

[242 rows x 8 columns]

8. In this question, you're going to create some plots to show the distribution of GDP per capita for the year 2020. Go to the Python Graph Gallery (https://www.python-graph-gallery.com/) and look at the different types of plots under the Distribution section. Create a histogram, a density plot, a boxplot, and a violin plot. What do you notice when you look at these plots? How do the plots compare and what information can you get out of one type that you can't necessarily get out of the others?

8a. Create a histogram

In [90]:
x = gdp_df.rename(columns = {'Country or Area':'Country','Value':'GDP_Per_Capita'})

In [91]:
my_list = [2020]

In [99]:
gdp_2020 = x.loc[x.Year.isin(my_list)]


In [138]:
gdp_graph =gdp_2020.drop(columns = ['Country','Year'])

In [152]:
gdp_graph

GDP_Per_Capita
1        1968.341002
22       3455.023119
55       3957.933804
88      13278.369769
121     10844.770764
...              ...
7503    10450.622382
7536     5402.538773
7565    16204.169107
7598     3183.650773
7631     1990.319419

[242 rows x 1 columns]